### PR TITLE
Cache + Resources

### PR DESCRIPTION
**Pull request by ignapas**
## What do these changes do?

* New widget ActivityManager **disabled**
  * Listing of computational services
  * Filtering capabilities
* Tags filter refactor
* Autocomplete field
* Dashboard events/display transitions fixes
* ResourceFactory refactor
* Store refactor

## Related issue number

#502 #525 #1058 

## How to test

Test that the interactions that require communication with the server still work.
Use it and improve it in your future development.

### CODE REVIEW

File: services/web/client/source/class/qxapp/component/filter/TagsFilter.js [hunk 1]
```diff
@@ -22,30 +22,30 @@ qx.Class.define("qxapp.component.filter.TagsFilter", {
...
-    this.__dropDown = new qx.ui.toolbar.MenuButton(this.tr("Tags"));
-    this.__dropDown.setMenu(this.__buildMenu());
-    this._add(this.__dropDown);
+    this._dropDown = new qx.ui.toolbar.MenuButton(label);
+    this._add(this._dropDown);
   },
 
   members: {
-    __dropDown: null,
+    _dropDown: null,
```
Comment (odeimaiz) [inline services/web/client/source/class/qxapp/component/filter/TagsFilter.js]: Why from two underscores to only one?

File: services/web/client/source/class/qxapp/component/filter/TagsFilter.js [hunk 2]
```diff
@@ -110,6 +91,24 @@ qx.Class.define("qxapp.component.filter.TagsFilter", {
       delete this.__tagButtons[tagName];
       // Dispatch
       this._filterChange(this.__activeTags);
+    },
+
+    _addOption: function(tagName) {
```
Comment (odeimaiz) [inline services/web/client/source/class/qxapp/component/filter/TagsFilter.js]: Same here. We use one underscore for protected and two for private

File: services/web/client/source/class/qxapp/component/form/renderer/PropForm.js [hunk 3]
```diff
@@ -348,7 +348,7 @@ qx.Class.define("qxapp.component.form.renderer.PropForm", {
         if (node1 && node2) {
           const port1 = node1.getOutput(port1Id);
           const port2 = node2.getInput(port2Id);
-          return qxapp.store.Store.getInstance().arePortsCompatible(port1, port2);
+          return qxapp.utils.Services.arePortsCompatible(port1, port2);
```
Comment (odeimaiz) [inline services/web/client/source/class/qxapp/component/form/renderer/PropForm.js]: Nice, much better!

Comment (odeimaiz) [inline services/web/client/source/class/qxapp/component/form/renderer/PropForm.js]: Nice, much better!

File: services/web/client/source/class/qxapp/component/filter/TagsFilter.js (hunk 2 above)
Comment (ignapas) [reply services/web/client/source/class/qxapp/component/filter/TagsFilter.js]: This one is used in child class StudyFilter

File: services/web/client/source/class/qxapp/component/filter/TagsFilter.js (hunk 1 above)
Comment (ignapas) [reply services/web/client/source/class/qxapp/component/filter/TagsFilter.js]: I thought it would be useful in derived classes

Comment (odeimaiz) [reply services/web/client/source/class/qxapp/component/filter/TagsFilter.js]: I can't see where else it's used

**Comment by odeimaiz** [comment]
Please, add documentation at least to the new classes and their constructors.

**Comment by odeimaiz** [comment]
There is a bug when building the data-tree. Datasets from S3 are used for populating both trees, S3 and Dat-Core:
image

File: services/web/client/source/class/qxapp/component/filter/TagsFilter.js (hunk 1 above)
Comment (ignapas) [reply services/web/client/source/class/qxapp/component/filter/TagsFilter.js]: It's not used. But it should be accessible to child classes..

**Comment by ignapas** [comment]
> There is a bug when building the data-tree. Datasets from S3 are used for populating both trees, S3 and Dat-Core:
> image

This should be fixed now

**Comment by ignapas** [comment]
> Please, add documentation at least to the new classes and their constructors.

I am on it

**Comment by ignapas** [comment]
> Is the filtering UX the same for all catalogs?
> how does cache gets invalidated? Can user do that?
> Can you put some screenshots of the views?

There are no visible changes involved in this PR. Even if there is a part of the ongoing development on the activity manager in here, it is deactivated and will become reviewable in a future PR.
To invalidate the cache for a particular resource, the cache can be cleared in this way:
``qxapp.store.Store.getInstance().reset("resourceName");``
Subsequent get calls for that resource will trigger a fetch from the server.
Also a call to the server can be forced, skipping the check in cache, with the dedicated parameter in the ``get`` and ``getOne`` methods.

**Pull request merged**